docs/source/handlers.rst

```diff
@@ -1,33 +1,42 @@
 Response Handlers
 =================
 
-Responses handlers are code which control the evaluation of the
-HTTP response in each test. For example the `response_strings` key
-described in :doc:`format` is a response handler. In many cases the
-default handlers described there will be sufficient for creating
-useful tests. When they are not, it is possible to add custom
-handlers by subclassing :class:`~gabbi.handlers.ResponseHandler` and
-adding that class in a call to :func:`~gabbi.driver.build_tests`::
+Response handlers determine how an HTTP response is being processed and checked
+against the expected result. For each entry starting with `response_`, the
+respective class is invoked with the corresponding values:
+
+    response_strings:
+        - "lorem ipsum"
+        - "dolor sit amet"
+
+This creates an instance of `StringResponseHandler`, passing the value
+`["lorem ipsum", "dolor sit amet"]` - it is then up to the response handler's
+implementation to interpret both the actual response and the expected values,
+thus determining whether the test passes or fails.
+
+While the default handlers (as described in :doc:`format`) are sufficient for
+most cases, it is possible to register additional custom handlers by passing a
+subclass of :class:`~gabbi.handlers.ResponseHandler` to
+:func:`~gabbi.driver.build_tests`::
 
     driver.build_tests(test_dir, loader, host=None,
                        intercept=simple_wsgi.SimpleWsgi,
-                       response_handlers=[TestResponseHandler])
+                       response_handlers=[MyResponseHandler])
 
 A subclass needs to define at least three things:
 
-* ``test_key_suffix``: This, along with the prefix ``response_`` forms
+* ``test_key_suffix``: This, along with the prefix ``response_``, forms
   the key used in the test structure. It is a class level string.
-* ``test_key_value``: The default value for this key if there is
-  nothing there. This needs to be an empty list or empty dict. It is
-  also a class level value.
-* ``action``: A method in the class which tests the expected values
-  against the HTTP response. A loop is made across each entry in the
-  test key with data passed to action:
-  If ``test_key_value`` is a list, then the arguments to``action`` are (in order):
-
-  * ``self``: The current object.
+* ``test_key_value``: The key's default value, either an empty list or empty
+  dict. It is also a class level value.
+* ``action``: An instance method which tests the expected values
+  against the HTTP response - it is invoked for each entry, with the parameters
+  depending on the default value. If ``test_key_value`` is a list, then the
```

Comment:
If we is "if this is a list" then we have these args, then there must be a "if this is a dict" then we have these args at the same level, you can't bury the if a dict in the "then we have" part.

If on the other hand you say, there are the args but if it is a list this one isn't used, it reflects reality and has grammatical parallelism.